### DISCUSSION TITLE
Are there any plans for a General Project Monitoring API

### DISCUSSION THREAD

**Discussion by colinkiama** [elementary/code#1392] [Q&A] [answered]
Currently you can monitor changes to Git repository projects using `MonitoredRespository`  and `GitManager`.

However, there doesn't seem to be an API for monitoring changes to non-git projects. This would be pretty useful for #134. It would make it easy to update search indexes when there are file changes to projects.

Are there any plans for this?

**Comment by jeremypw** [comment, marked as answer]
Each `Scratch.Foldermanager.Folderitem` has a `GLib.FileMonitor` connected to an `on_changed ()` handler.  Maybe this could be used to invalidate the search index?

**Reply by colinkiama** [reply]
Thanks, I have an idea of what I could do now
